src/copy.c

```diff
@@ -554,28 +563,26 @@ timescaledb_CopyGetAttnums(TupleDesc tupDesc, Relation rel, List *attnamelist)
 }
 
 static void
-copy_security_check(Relation rel, List *attnums)
+copy_constraints_and_check(ParseState *pstate, Relation rel, List *attnums)
 {
-	List *range_table = NIL;
 	ListCell *cur;
-	RangeTblEntry *rte;
 	char *xactReadOnly;
-
-	rte = makeNode(RangeTblEntry);
-	rte->rtekind = RTE_RELATION;
-	rte->relid = RelationGetRelid(rel);
-	rte->relkind = rel->rd_rel->relkind;
+#if PG12_GE
```

Comment:
It would be good to split this into 2 functions and make the addition of the range table entry a separate function and call that explicitly.